src/compositions/useBlockDocument.ts

```diff
@@ -1,29 +1,32 @@
 import { useSubscriptionWithDependencies } from '@prefecthq/vue-compositions'
-import { computed, ref, Ref } from 'vue'
+import { computed, MaybeRefOrGetter, toRef, toValue } from 'vue'
 import { useCan } from '@/compositions/useCan'
 import { useWorkspaceApi } from '@/compositions/useWorkspaceApi'
 import { WorkspaceBlockDocumentsApi } from '@/services/WorkspaceBlockDocumentsApi'
+import { Getter } from '@/types/reactivity'
 import { UseEntitySubscription } from '@/types/useEntitySubscription'
 
 export type UseBlockDocument = UseEntitySubscription<WorkspaceBlockDocumentsApi['getBlockDocument'], 'blockDocument'>
 
-export function useBlockDocument(blockDocumentId: string | Ref<string | null | null>): UseBlockDocument {
-  const id = ref(blockDocumentId)
+export function useBlockDocument(blockDocumentId: MaybeRefOrGetter<string | null | null>): UseBlockDocument {
   const api = useWorkspaceApi()
   const can = useCan()
 
-  const parameters = computed<[string] | null>(() => {
-    if (!id.value) {
+  const getter: Getter<[string] | null> = () => {
+    if (!can.read.block) {
       return null
     }
```

Comment:
I'm assuming the switch in checks for this are to avoid making an extra `toValue` call that we don't need to?

Reply:
Correct, and I think it makes sense in general to do permission checks first.